lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -4052,6 +4067,45 @@ static void diagnoseConformanceFailure(Type T,
     }
   }
 
+  // One cannot meaningfully declare conformance to the NSObject protocol
+  // in Swift. Suggest inheritance from NSObject instead.
+  if (isNSObjectProtocol(Proto)) {
+    if (T->getClassOrBoundGenericClass()) {
```

Comment:
I'm not familiar with the surrounding code—does this call return the same class you `dyn_cast<ClassDecl>()` to get later?